core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -106,6 +106,28 @@ case class SparkListenerExecutorRemoved(time: Long, executorId: String, reason:
   extends SparkListenerEvent
 
 @DeveloperApi
+case class SparkListenerExecutorBlacklisted(
+    time: Long,
+    executorId: String,
+    taskFailures: Int)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerExecutorUnblacklisted(time: Long, executorId: String)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerNodeBlacklisted(
+    time: Long,
+    nodeId: String,
```

Comment:
Hmm... I don't remember `nodeId` being a thing in Spark. Do you mean a hostname here, or ip address, or something else? Should probably use a better name, or at least document what it is.

(I see BlacklistTracker uses that, but as a user, I don't know what "nodeId" is.)